pkg/controller/master/rancher/pod_resources.go

```diff
@@ -0,0 +1,132 @@
+package rancher
+
+import (
+	"encoding/json"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// Since multi-cluster-management-agent is disabled in the Harvester embedded Rancher,
+// copy and modified this file from Rancher to fix https://github.com/harvester/harvester/issues/2310
+// source file: https://github.com/rancher/rancher/tree/release/v2.6/pkg/controllers/managementagent/podresources
+
+const (
+	RequestsAnnotation = "management.cattle.io/pod-requests"
+	LimitsAnnotation   = "management.cattle.io/pod-limits"
+)
+
+func (h *Handler) PodResourcesOnChanged(key string, node *corev1.Node) (*corev1.Node, error) {
+	if node == nil {
+		return node, nil
+	}
+
+	h.nodeController.EnqueueAfter(node.Name, 15*time.Second)
+
+	pods, err := h.getNonTerminatedPods(node)
+	if err != nil {
+		return nil, err
+	}
+
+	requests, limits, err := getPodResourceAnnotations(pods)
+	if err != nil {
+		return nil, err
+	}
+
+	if node.Annotations[RequestsAnnotation] != requests ||
+		node.Annotations[LimitsAnnotation] != limits {
+		toUpdate := node.DeepCopy()
+		if toUpdate.Annotations == nil {
```

Comment:
If really worry about `toUpdate.Annotations == nil `, then it should be before L38,  as `node.Annotations[RequestsAnnotation]` may cause panic when it's nil

Reply:
read nil map won't panic